cf-oscore/src/main/java/org/eclipse/californium/oscore/Encryptor.java

```diff
@@ -209,8 +213,15 @@ protected static Encrypt0Message prepareCOSEStructure(byte[] confidential, byte[
 			}
 		} else {
 			bRes.write(firstByte);
+		}
+		
+		//If the OSCORE option is length 1 and 0x00, it can be empty
+		//See https://tools.ietf.org/html/draft-ietf-core-object-security-16#page-33 point 4
+		byte[] optionBytes = bRes.toByteArray();
```

Comment:
The javadoc says "Creates a newly allocated byte array.".
So I'm wondering, why `toByteArray` is called twice.
For `new byte[0]` consider to use `org.eclipse.californium.elements.util.Bytes.EMPTY`

Reply:
Yes, it's not needed to call it twice. I have changed that and also use Bytes.EMPTY instead now. See commit 003a99433cfa418f3babef9ee7d1af7d2ae0c2ba.